### ISSUE TITLE
Build Slack bot to send event reminders hours before they happen

### ISSUE THREAD

**Issue by ericlathrop** [louisvilleio/louisville.io#19]
We have all the events in JSON, just need a cron job to find upcoming events and send them to slack.

**Comment by thebouv** [comment]
Suggestion on time interval for this?

1 hour before? 2 hours before?

I'll probably make it also just a request on command as well like:

```
loubot events
```

And it'll list the events happening today on demand (perhaps via DM, let me know preference, if you have one).

Also, perhaps I'm overlooking where it ends up, but what is the URL for the events JSON?

BTW, loubot is here at the moment: https://github.com/thebouv/loubot

I have him running on heroku in my own slack while I hack on his bits.

**Comment by ericlathrop** [comment]
There isn't a URL. The JSON ends up in the data/events.json file inside of the louisville.io repo. I was initially thinking this would be a tiny cron job that runs hourly on the louisville.io server, rather than a full-fledged bot that you can talk to interactively. I'm not arguing against an interactive bot, but it sounds like a lot more work.

**Comment by thebouv** [comment]
There are several ways to do this.  For sending data into our slack, we can do a incoming webhook -- which counts as an integration against our limit.

If we then want other integrations using incoming or outgoing webhooks, they could each count against said limit.

Having a hubot integration, however, allows the bot to do lots of stuff for us, including all the incoming/outgoing web data we want.

Basically, with the Slack APIs, there's more than one way to do it.

Weather was mentioned, code of conduct posting was mentioned, and we currently use a giphy integration as well.  All of which can be done by a single hubot integration.  I think the "free" slack is limited to 5 integrations total.

I'm fine with whatever you want done. I'm just playing around with this code for hubot/loubot.

**Comment by ericlathrop** [comment]
I'm fine with Hubot if you want to implement it. I would like it to be deployable with [our ansible playbooks](https://github.com/louisvilleio/louisville.io-ansible) on the same server we're already using (DigitalOcean, debian jesse)

**Comment by thebouv** [comment]
It should be able to do that. It's just node with CoffeeScript.

I don't know Ansible at all, but no time like the present to learn.

I'll get the bot to a functioning state and figure it out from there.

Or if I hit a wall, I'll go simpler, do just the events thing as something else with another part of Slack's API that can just run via cron.

Fun to learn the ins and outs no matter what.

Speaking of cron, though, no matter what direction I go:  ideas on when it should send out the messages? 1 hour before? 2 hours before? First message of the day in #general?

**Comment by ericlathrop** [comment]
Instead of "x hours before" I like the idea of sending the messages at 
* 10am for things starting noon or later the same day
* 3pm for things starting 5pm or later the same day
* 3pm for events starting tomorrow
I've been thinking of using a similar schedule for the twitter bot (#14)

**Comment by thebouv** [comment]
:thumbsup:
